src/hooks/useProducts.ts

```diff
@@ -0,0 +1,33 @@
+import { keepPreviousData, useQuery } from "@tanstack/react-query";
+import axios from "axios";
+
+const getProducts = async (
```

Comment:
suggestion move the api call to `src/api/products.ts` to follow our projects standards and also this structure ensures your codebase stays organized as it grows and api functions in api can be reused across different hooks or components